src/ServiceControl.Monitoring/Timings/TimingsStore.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace ServiceControl.Monitoring.Timings
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Processing.RawData.NServiceBus.Metrics;
+
+    public abstract class TimingsStore
+    {
+        ConcurrentDictionary<string, ConcurrentDictionary<DateTime, MeasurementInterval>> timings = 
+            new ConcurrentDictionary<string, ConcurrentDictionary<DateTime, MeasurementInterval>>();
+
+        public void Store(string endpointName, LongValueOccurrences message, DateTime now)
+        {
+            var endpointData = timings.GetOrAdd(endpointName, new ConcurrentDictionary<DateTime, MeasurementInterval>());
+
+            for (var i = 0; i < message.Ticks.Length; i++)
+            {
+                var date = new DateTime(message.BaseTicks + message.Ticks[i], DateTimeKind.Utc);
+                var intervalId = IntervalId(date);
+
+                endpointData.AddOrUpdate(
+                    intervalId,
+                    _ => new MeasurementInterval(1, message.Values[i]),
+                    (_, b) =>
+                    {
+                        b.TotalMeasurements += 1;
+                        b.TotalTime += message.Values[i];
+                        return b;
+                    });
+            }
+
+            if (endpointData.Count > 2 * NumberOfHistoricalIntervals)
+            {
+                var validIntervals = GenerateIntervalIds(now, NumberOfHistoricalIntervals);
+
+                foreach (var interval in endpointData.Keys)
+                {
+                    if (validIntervals.Contains(interval) == false)
+                    {
+                        MeasurementInterval bucket;
+                        endpointData.TryRemove(interval, out bucket);
+                    }
+                }
+
+            }
```

Comment:
The data is stored only for non-empty intervals so just counting the intervals will not do the trick I'm afraid. 
When it comes to passing `now` it also enables writing predictable [tests](https://github.com/Particular/ServiceControl.Monitoring/pull/21/files#diff-8c25298ee0255593d53dc1673450ce55R11).